sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQueryStatusAndProgressSuite.scala

```diff
@@ -171,6 +174,47 @@ class StreamingQueryStatusAndProgressSuite extends StreamTest {
       query.stop()
     }
   }
+
+  test("SPARK-19378: Continue reporting stateOp and eventTime metrics even if there is no data") {
+    import testImplicits._
+
+    withSQLConf(SQLConf.STREAMING_NO_DATA_PROGRESS_EVENT_INTERVAL.key -> "10") {
+      val inputData = MemoryStream[(Int, String)]
+
+      val query = inputData.toDS().toDF("value", "time")
+        .select('value, 'time.cast("timestamp"))
+        .withWatermark("time", "10 seconds")
+        .groupBy($"value")
+        .agg(count("*"))
+        .writeStream
+        .queryName("metric_continuity")
+        .format("memory")
+        .outputMode("complete")
+        .start()
+      try {
+        inputData.addData((1, "2017-01-26 01:00:00"), (2, "2017-01-26 01:00:02"))
+        query.processAllAvailable()
+
+        val progress = query.lastProgress
+        assert(progress.eventTime.size() > 1)
+        assert(progress.stateOperators.length > 0)
+        // Should emit new progresses every 10 ms, but we could be facing a slow Jenkins
+        eventually(timeout(1 minute)) {
+          val nextProgress = query.lastProgress
+          assert(nextProgress.timestamp !== progress.timestamp)
+          assert(nextProgress.numInputRows === 0)
+          assert(nextProgress.eventTime.get("min") === "2017-01-26 01:00:00")
```

Comment:
This does not make sense. if there is no data in the last trigger, the min, max, avg timestamps cannot be different. 
and what about the watermark?